### DISCUSSION TITLE
Are there a way to update Text WebParts in a page?

### DISCUSSION THREAD

**Discussion by mohammadamer** [pnp/pnpcore#1356] [General] [answered]
Are there a way to update Text WebParts in a page?

I've been checking the [documentation ](https://pnp.github.io/pnpcore/using-the-sdk/pages-webparts.html) and I found how to add a text webpart to a page, but I didn't find a way how to edit the text of text webparts in a page.
Thanks in advance.

**Comment by jansenbe** [comment, marked as answer]
@mohammadamer : you can cast the control you need to `IPageText` and then set the `Text` property. This test (https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core.Test/SharePoint/PagesTests.cs#L379-L415) shows that, but since it's a test it uses the internal only `PageText`, you need to use `IPageText`

**Reply by mohammadamer** [reply]
Thanks a lot @jansenbe :)

**Comment by magdalenafernlund-sherpas** [comment]
As of [this update](https://www.microsoft.com/en-us/microsoft-365/roadmap?rtc=1&searchterms=386904&filters=&searchterms=386904?rtc=1&searchterms=386904&filters=&searchterms=386904), the banner webpart seems to have introduced some strange behaviour, which seems to have been resolved by updating Pnp Core to 1.14. 

@jansenbe Is there a way to update the text inside the banner, like you suggested here to set the text property on the IPageText. Tried to use IPageWebPart, but can't seems to figure out how to update the text.

**Reply by jansenbe** [reply]
@magdalenafernlund-sherpas : this should have been fixed in the latest builds
